pkg/kubectl/cmd/util/factory.go

```diff
@@ -214,13 +216,25 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			}
 			return kubectl.ReaperFor(mapping.Kind, client)
 		},
-		Validator: func(validate bool) (validation.Schema, error) {
+		Validator: func(validate bool, cacheDir string) (validation.Schema, error) {
 			if validate {
 				client, err := clients.ClientForVersion("")
 				if err != nil {
 					return nil, err
 				}
-				return &clientSwaggerSchema{client, client.ExperimentalClient, api.Scheme}, nil
+				version, err := client.ServerVersion()
```

Comment:
There's no other good way to do it.  I guess we could do it per-host, but the version on the host can change (upgrades) or the host address could change (dynamic IP)